Include/internal/pycore_object.h

```diff
@@ -466,8 +466,8 @@ static inline void _PyObject_GC_TRACK(
     PyGC_Head *last = (PyGC_Head*)(generation0->_gc_prev);
     _PyGCHead_SET_NEXT(last, gc);
     _PyGCHead_SET_PREV(gc, last);
-    /* Young objects will be moved into the visited space during GC, so set the bit here */
-    gc->_gc_next = ((uintptr_t)generation0) | (uintptr_t)interp->gc.visited_space;
+    uintptr_t not_visited = 1 ^ interp->gc.visited_space;
+    gc->_gc_next = ((uintptr_t)generation0) | not_visited;
```

Comment:
I don't think the comment describing the `_gc_next` field in the header file is correct:

```
    // Pointer to next object in the list.
    // 0 means the object is not tracked
    uintptr_t _gc_next;
```

Reply:
That comment is still correct.

We only set the low bits of the `_gc_next` field if the object is tracked.
`_gc_next == 0` still implies the object is not tracked. https://github.com/python/cpython/blob/main/Include/internal/pycore_gc.h#L82

Reply:
But if it's tracked it's not exactly the pointer.

Reply:
Ah you mean the "Pointer to next object in the list", not the "0 means the object is not tracked"?

How about?
```C
// Tagged pointer to next object in the list.
// 0 means the object is not tracked
uintptr_t _gc_next;
```

Reply:
Ok. 

Where are tagged pointers documented?

Reply:
In the "Optimization: delay tracking containers" section of garbage_collector.md

More generally, https://en.wikipedia.org/wiki/Tagged_pointer